examples/connection_reuse.rs

```diff
@@ -5,16 +5,14 @@ use std::io;
 
 fn main() -> Result<()> {
     let args = std::env::args();
-    let url: Url = args
+    let host = args
         .skip(1)
         .next()
-        .unwrap_or("http://www.google.com".into())
-        .parse()?;
+        .unwrap_or("http://www.google.com".into());
 
     let mut client = HttpClient::<std::net::TcpStream>::new();
     for path in &["/", "/favicon.ico", "/robots.txt"] {
-        let mut url = url.clone();
-        url.path = path.parse()?;
+        let url = Url::parse(format!("{}{}", host, path).as_str())?;
```

Comment:
instead of doing this, could you instead parse URL outside of the loop and use `Url::push`. `path` here isn't event a path and actually could just be `Option<String>` to represent adding a component.

I like to avoid doing any string concatenation for path-like things (even if you can prove it can't cause issues in this case) since it can lead to issues like double `/` or missing `/`

Reply:
done by switch to url.set_path